blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/utils/PlotRequestParser.java

```diff
@@ -89,11 +89,23 @@ public static RollupsQueryParams parseParams(Map<String, List<String>> params) t
             throw new InvalidRequestException("Invalid parameter: to="+ to);
         }
 
-        long fromTime = Long.parseLong(from.get(0));
-        long toTime = Long.parseLong(to.get(0));
+        long fromTime;
+
+        try {
+            fromTime = Long.parseLong(from.get(0));
+        } catch (NumberFormatException e) {
+            throw new InvalidRequestException("parameter 'from' must be a valid long");
+        }
+
+        long toTime;
+        try {
+            toTime = Long.parseLong(to.get(0));
+        } catch (NumberFormatException e) {
+            throw new InvalidRequestException("parameter 'to' must be a valid long");
+        }
 
         if (toTime <= fromTime) {
-            throw new InvalidRequestException("paramter 'to' must be greater than 'from'");
+            throw new InvalidRequestException("parameter 'to' must be greater than 'from'");
```

Comment:
Nice to have: it's usually good to print the offending value in the message. But I know it's a pain :-)